PermissionDenied error due to metadata folder

My `~/projects/` folder is not a normal directory, as it is technically a hard drive set to automatically mount to that folder on login. I use a main SSD, and a hard-drive for the `projects` folder to be able to get fast boot times, and a large amount of storage.

Targeting this directory with `timetrack` results in: 
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Os { code: 13, kind: PermissionDenied, message: "Permission denied" })', libcore/result.rs:945:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:227
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:511
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:426
   6: rust_begin_unwind
             at libstd/panicking.rs:337
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:92
   8: core::result::unwrap_failed
   9: timetrack::track::<impl timetrack::TimeTracker<'a>>::track
  10: timetrack::main
  11: std::rt::lang_start::{{closure}}
  12: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:310
  13: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:105
  14: std::rt::lang_start_internal
             at libstd/panicking.rs:289
             at libstd/panic.rs:392
             at libstd/rt.rs:58
  15: main
  16: __libc_start_main
  17: _start
```

In my `projects` folder, there is a file named `lost+found`, which is created for EXT4 drives for FS operations (due to a hard drive being mounted there). Changing this directory to a specific project folder results in the program working as expected.

As a feature request, can there be the option to avoid folders with `root` permission (which seems like a better solution), or maintaining an internal list of folder names to avoid (which seems like a good idea, but I don't know whether this would work in practice).

I run Arch Linux x86_64, updated to latest packages (as of current writing), using the `cargo install timetrack` version of the application (0.2.5).

Any help would be greatly appreciated!

Currently, it still seems like I can't build `timetrack` - I get this compilation error:

```
[permission-denied-#2][~/projects/timetrack]$ cargo run --verbose -- track
       Fresh version_check v0.1.4
       Fresh winapi-build v0.1.1
       Fresh cfg-if v0.1.5
       Fresh winapi v0.2.8
       Fresh unicode-xid v0.1.0
       Fresh libc v0.2.43
       Fresh bitflags v0.4.0
       Fresh ucd-util v0.1.1
       Fresh quick-error v1.2.2
       Fresh bytes v0.3.0
       Fresh byteorder v1.2.6
       Fresh same-file v1.0.2
       Fresh rustc-serialize v0.3.24
       Fresh slab v0.1.3
       Fresh unicode-width v0.1.5
       Fresh utf8-ranges v1.0.1
       Fresh bitflags v1.0.3
       Fresh encode_unicode v0.3.2
       Fresh vec_map v0.8.1
       Fresh ansi_term v0.11.0
       Fresh bitflags v0.7.0
       Fresh strsim v0.7.0
       Fresh termcolor v1.0.3
       Fresh log v0.4.3
       Fresh net2 v0.2.33
       Fresh memchr v2.0.2
       Fresh time v0.1.40
       Fresh memchr v1.0.2
       Fresh inotify v0.3.0
       Fresh filetime v0.1.15
       Fresh atty v0.2.11
       Fresh directories v1.0.1
       Fresh nix v0.5.1
       Fresh regex-syntax v0.6.2
       Fresh humantime v1.1.1
       Fresh term v0.5.1
       Fresh walkdir v2.2.0
       Fresh textwrap v0.10.0
       Fresh proc-macro2 v0.4.13
       Fresh log v0.3.9
       Fresh aho-corasick v0.6.8
       Fresh csv v0.15.0
       Fresh serde v1.0.72
       Fresh clap v2.32.0
       Fresh lazy_static v1.1.0
       Fresh kernel32-sys v0.2.2
       Fresh ws2_32-sys v0.2.1
       Fresh quote v0.6.6
       Fresh toml v0.4.6
       Fresh thread_local v0.3.6
       Fresh prettytable-rs v0.7.0
       Fresh miow v0.1.5
       Fresh syn v0.14.8
       Fresh regex v1.0.4
       Fresh mio v0.5.1
       Fresh serde_derive v1.0.72
       Fresh env_logger v0.5.13
       Fresh notify v4.0.4
   Compiling timetrack v0.2.5 (file:///home/shavs/projects/timetrack)
     Running `rustc --crate-name timetrack src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=41b25a061af3674c -C extra-filename=-41b25a061af3674c --out-dir /home/shavs/projects/timetrack/target/debug/deps -C incremental=/home/shavs/projects/timetrack/target/debug/incremental -L dependency=/home/shavs/projects/timetrack/target/debug/deps --extern clap=/home/shavs/projects/timetrack/target/debug/deps/libclap-0a1cef982b608eea.rlib --extern directories=/home/shavs/projects/timetrack/target/debug/deps/libdirectories-fd7857fdd314e50a.rlib --extern env_logger=/home/shavs/projects/timetrack/target/debug/deps/libenv_logger-6eabe9fa1856a452.rlib --extern log=/home/shavs/projects/timetrack/target/debug/deps/liblog-3df402b02e823472.rlib --extern notify=/home/shavs/projects/timetrack/target/debug/deps/libnotify-d21a55077c111772.rlib --extern prettytable=/home/shavs/projects/timetrack/target/debug/deps/libprettytable-27c4c9e64b14091a.rlib --extern serde=/home/shavs/projects/timetrack/target/debug/deps/libserde-07ff9064c8d17151.rlib --extern serde_derive=/home/shavs/projects/timetrack/target/debug/deps/libserde_derive-a3b99743f6e8c9de.so --extern toml=/home/shavs/projects/timetrack/target/debug/deps/libtoml-36b8237e6015e2cb.rlib`
error[E0432]: unresolved import `notify::FsEventWatcher`
 --> src/watcher/mod.rs:5:5
  |
5 | use notify::FsEventWatcher;
  |     ^^^^^^^^^^^^^^^^^^^^^^ no `FsEventWatcher` in the root

warning: unused import: `std::fs`
 --> src/schedule/mod.rs:2:5
  |
2 | use std::fs;
  |     ^^^^^^^
  |
  = note: #[warn(unused_imports)] on by default

warning: unused import: `std::fs::OpenOptions`
 --> src/schedule/mod.rs:3:5
  |
3 | use std::fs::OpenOptions;
  |     ^^^^^^^^^^^^^^^^^^^^

warning: unused import: `std::io::Write`
 --> src/schedule/mod.rs:4:5
  |
4 | use std::io::Write;
  |     ^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: Could not compile `timetrack`.

Caused by:
  process didn't exit successfully: `rustc --crate-name timetrack src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=41b25a061af3674c -C extra-filename=-41b25a061af3674c --out-dir /home/shavs/projects/timetrack/target/debug/deps -C incremental=/home/shavs/projects/timetrack/target/debug/incremental -L dependency=/home/shavs/projects/timetrack/target/debug/deps --extern clap=/home/shavs/projects/timetrack/target/debug/deps/libclap-0a1cef982b608eea.rlib --extern directories=/home/shavs/projects/timetrack/target/debug/deps/libdirectories-fd7857fdd314e50a.rlib --extern env_logger=/home/shavs/projects/timetrack/target/debug/deps/libenv_logger-6eabe9fa1856a452.rlib --extern log=/home/shavs/projects/timetrack/target/debug/deps/liblog-3df402b02e823472.rlib --extern notify=/home/shavs/projects/timetrack/target/debug/deps/libnotify-d21a55077c111772.rlib --extern prettytable=/home/shavs/projects/timetrack/target/debug/deps/libprettytable-27c4c9e64b14091a.rlib --extern serde=/home/shavs/projects/timetrack/target/debug/deps/libserde-07ff9064c8d17151.rlib --extern serde_derive=/home/shavs/projects/timetrack/target/debug/deps/libserde_derive-a3b99743f6e8c9de.so --extern toml=/home/shavs/projects/timetrack/target/debug/deps/libtoml-36b8237e6015e2cb.rlib` (exit code: 101)
```

It seems as though it can't use `FsEventWatcher`, which is odd, as I thought running `cargo run` or `cargo build` would bring down any missing dependencies? This was after running `cargo run -- track`, then adding the `--verbose` flag afterwards to get a bigger chunk of debug.

--- a/src/config/mod.rs
+++ b/src/config/mod.rs
@@ -11,6 +11,8 @@ use std::path::Path;
 use std::path::PathBuf;
 use toml;
 use TimeTracker;
+use watcher;
+use std::sync::mpsc::channel;
 
 pub struct Configuration {
     user_config_path: PathBuf, // this file should not be read outside this module
@@ -78,6 +80,19 @@ pub fn get_config() -> Configuration {
 impl<'a> TimeTracker<'a> {
     pub fn print_config(&self) {
         println!("{}", self.config);
+        println!("Starting self test..");
+        let (tx, _rx) = channel();
+        for track_path in &self.config.track_paths {
+            match watcher::get_watcher(track_path, tx.clone()) {
+                Ok(_) => {
+                    println!("Successfully added watcher for path {}", track_path.to_string_lossy());
+                },
+                Err(err) => {
+                    println!("Error {} adding watcher for path {:?}", err, track_path.to_string_lossy());
+                },
+            };
+        }
+        println!("Completed self test.");
     }
 }
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -13,6 +13,7 @@ mod clear;
 mod config;
 mod schedule;
 mod track;
+mod watcher;
 
 use config::Configuration;
 

--- a/src/track/mod.rs
+++ b/src/track/mod.rs
@@ -1,4 +1,4 @@
-use notify::{DebouncedEvent, RecommendedWatcher, RecursiveMode, Watcher};
+use notify::DebouncedEvent;
 use std::collections::HashSet;
 use std::fs::OpenOptions;
 use std::io::Write;
@@ -11,9 +11,6 @@ use std::time::Duration;
 use std::time::Instant;
 use std::time::SystemTime;
 use TimeTracker;
-use std::sync::mpsc::Sender;
-use notify::FsEventWatcher;
-use notify;
 use watcher;
 
 impl<'a> TimeTracker<'a> {

--- a/src/track/mod.rs
+++ b/src/track/mod.rs
@@ -11,6 +11,10 @@ use std::time::Duration;
 use std::time::Instant;
 use std::time::SystemTime;
 use TimeTracker;
+use std::sync::mpsc::Sender;
+use notify::FsEventWatcher;
+use notify;
+use watcher;
 
 impl<'a> TimeTracker<'a> {
     pub fn track(&self) {
@@ -19,11 +23,10 @@ impl<'a> TimeTracker<'a> {
         let mut watchers = vec![]; // need to keep ownership of watchers so they aren't dropped at end of for-loop
 
         for track_path in &self.config.track_paths {
-            let mut watcher: RecommendedWatcher =
-                Watcher::new(tx.clone(), Duration::from_secs(0)).unwrap();
-            watcher.watch(track_path, RecursiveMode::Recursive).unwrap();
-
-            watchers.push(watcher);
+            match watcher::get_watcher(track_path, tx.clone()) {
+                Ok(watcher) => watchers.push(watcher),
+                Err(_) => {}, // errors are silent here, but reported by timetrack config
+            }
         }
 
         let mut first_record_time;

--- a/src/watcher/mod.rs
+++ b/src/watcher/mod.rs
@@ -2,10 +2,9 @@ use notify::{DebouncedEvent, RecommendedWatcher, RecursiveMode, Watcher};
 use std::path::PathBuf;
 use std::time::Duration;
 use std::sync::mpsc::Sender;
-use notify::FsEventWatcher;
 use notify;
 
-pub fn get_watcher(track_path: &PathBuf, tx: Sender<DebouncedEvent>) -> Result<FsEventWatcher, notify::Error> {
+pub fn get_watcher(track_path: &PathBuf, tx: Sender<DebouncedEvent>) -> Result<RecommendedWatcher, notify::Error> {
     let mut watcher: RecommendedWatcher =
         Watcher::new(tx, Duration::from_secs(0))?;
     watcher.watch(track_path, RecursiveMode::Recursive)?;

--- a/src/watcher/mod.rs
+++ b/src/watcher/mod.rs
@@ -1,16 +1,6 @@
 use notify::{DebouncedEvent, RecommendedWatcher, RecursiveMode, Watcher};
-use std::collections::HashSet;
-use std::fs::OpenOptions;
-use std::io::Write;
-use std::path::Path;
 use std::path::PathBuf;
-use std::sync::mpsc::channel;
-use std::sync::mpsc::TryRecvError;
-use std::thread;
 use std::time::Duration;
-use std::time::Instant;
-use std::time::SystemTime;
-use TimeTracker;
 use std::sync::mpsc::Sender;
 use notify::FsEventWatcher;
 use notify;

--- a/src/watcher/mod.rs
+++ b/src/watcher/mod.rs
@@ -1,0 +1,24 @@
+use notify::{DebouncedEvent, RecommendedWatcher, RecursiveMode, Watcher};
+use std::collections::HashSet;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::path::Path;
+use std::path::PathBuf;
+use std::sync::mpsc::channel;
+use std::sync::mpsc::TryRecvError;
+use std::thread;
+use std::time::Duration;
+use std::time::Instant;
+use std::time::SystemTime;
+use TimeTracker;
+use std::sync::mpsc::Sender;
+use notify::FsEventWatcher;
+use notify;
+
+pub fn get_watcher(track_path: &PathBuf, tx: Sender<DebouncedEvent>) -> Result<FsEventWatcher, notify::Error> {
+    let mut watcher: RecommendedWatcher =
+        Watcher::new(tx, Duration::from_secs(0))?;
+    watcher.watch(track_path, RecursiveMode::Recursive)?;
+
+    Ok(watcher)
+}

